### DISCUSSION TITLE
How to wait specific document state complete after user invoked extra data load on the webpage

### DISCUSSION THREAD

**Discussion by assopri** [cefsharp/CefSharp#3675] [General]
Let's say we have a webpage, where user need to scroll down for getting more data uploaded (usually we can meet such behavior at social networks like Facebook and Instagram). How can I identify the moment when loading extra data is finished (to proceed executing other actions on the webpage or just extract all data, including extra loaded)?

**Comment by amaitland** [comment]
I'm not aware of a one size fits all solution. Each website behaves differently. 

Perhaps someone on https://stackoverflow.com/ might have some suggestions.

**Reply by assopri** [reply]
Ok, maybe I can just hook to some document state changed event? Browser.LoadingStateChanged += handler;

**Reply by amaitland** [reply]
LoadingStateChange will be called when all resources defined in the html are loaded. Dynamically loaded content like delay loaded content will happen after this.

**Reply by assopri** [reply]
maybe then I can just track Browser.IsLoading? If it's false = I'm still waiting, true = I consider loading is completed.

**Reply by amaitland** [reply]
https://github.com/cefsharp/CefSharp/blob/cefsharp/91/CefSharp.WinForms/ChromiumWebBrowser.cs#L677
